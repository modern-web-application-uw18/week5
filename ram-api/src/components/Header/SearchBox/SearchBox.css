.SearchBox {
    
}

.input-group {
    width: 220px;
    opacity: .6;
    transition: opacity 2.8s;
}

.input-group:hover,
.input-group:active,
.input-group.active {
    opacity: 1;
    transition: opacity 0.4s;
}