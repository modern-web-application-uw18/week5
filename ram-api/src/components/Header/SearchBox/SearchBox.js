import React, { Component } from 'react';
import './SearchBox.css';
import PropTypes from 'prop-types';

class SearchBox extends Component {

    state = {
        searchBoxValue: ''
    }


    handleChange = (e) => {
        let value = e.target.value;
        this.setState((prevState) => {
            this.props.searchHandler(value);
            return { searchBoxValue: value}
        });    
    }


    render() {

        return (
            <div className="SearchBox">
                <div className="row">
                    <div className="col-md-12">
                        <div className="input-group">
                            <input type="text" className="form-control" placeholder="Filter by character name..." value={this.state.searchBoxValue} onChange={this.handleChange} />
                        </div>
                    </div>
                </div>

            </div>
        )
    }
}



SearchBox.propTypes = {
    searchHandler: PropTypes.func.isRequired
}

export default SearchBox;