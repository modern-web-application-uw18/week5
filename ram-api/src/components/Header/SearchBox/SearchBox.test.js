import React from 'react';
import { mount } from 'enzyme';
import SearchBox from './SearchBox'

describe('<SearchBox />', () => {
    it('Should render SearchBox', () => {

        const wrapper = mount(  <SearchBox searchHandler={jest.fn()} />      );


        expect(wrapper).toMatchSnapshot();
    })


    it('Should simulate onchange event', () => {

        const wrapper = mount(  <SearchBox searchHandler={jest.fn()} />      );


        wrapper.find('.form-control').simulate('change');
        expect(wrapper).toMatchSnapshot();
    })



    it('Should simulate componentWillUnmount', () => {

        const wrapper = mount(  <SearchBox searchHandler={jest.fn()} />      );


        wrapper.unmount();

        expect(wrapper).toMatchSnapshot();
    })


})




