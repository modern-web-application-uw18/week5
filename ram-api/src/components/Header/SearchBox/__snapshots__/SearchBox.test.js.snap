// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBox /> Should render SearchBox 1`] = `
<SearchBox
  searchHandler={[Function]}
>
  <div
    className="SearchBox"
  >
    <div
      className="row"
    >
      <div
        className="col-md-12"
      >
        <div
          className="input-group"
        >
          <input
            className="form-control"
            onChange={[Function]}
            placeholder="Filter by character name..."
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</SearchBox>
`;

exports[`<SearchBox /> Should simulate componentWillUnmount 1`] = `""`;

exports[`<SearchBox /> Should simulate onchange event 1`] = `
<SearchBox
  searchHandler={[Function]}
>
  <div
    className="SearchBox"
  >
    <div
      className="row"
    >
      <div
        className="col-md-12"
      >
        <div
          className="input-group"
        >
          <input
            className="form-control"
            onChange={[Function]}
            placeholder="Filter by character name..."
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</SearchBox>
`;
